checkov/terraform/checks/data/aws/GithubActionsOIDCTrustPolicy.py

```diff
@@ -1,59 +1,85 @@
 from checkov.common.util.type_forcers import force_list
 from checkov.terraform.checks.data.base_check import BaseDataCheck
-
-gh_repo_regex = re.compile(r'repo:[^/]+/[^/]+')
+from checkov.common.util.oidc_utils import gh_abusable_claims, gh_repo_regex
 
 
 class GithubActionsOIDCTrustPolicy(BaseDataCheck):
     def __init__(self):
-        name = 'Ensure GitHub Actions OIDC trust policies only allows actions from a specific known organization'
+        name = "Ensure GitHub Actions OIDC authorization policies only allow safe claims and claim order"
         id = "CKV_AWS_358"
         supported_data = ("aws_iam_policy_document",)
         categories = [CheckCategories.IAM]
         super().__init__(name=name, id=id, categories=categories, supported_data=supported_data)
 
     def scan_data_conf(self, conf: Dict[str, List[Any]]) -> CheckResult:
-        statements = force_list(conf.get('statement'))
+        statements = force_list(conf.get("statement"))
         for statement in statements:
             found_federated_gh_oidc = False
             if isinstance(statement, dict):
-                if statement.get('principals'):
-                    principals = statement['principals']
+                if statement.get("principals"):
+                    principals = statement["principals"]
                     for principal in force_list(principals):
-                        if 'type' not in principal and 'identifiers' not in principal:
+                        if "type" not in principal and "identifiers" not in principal:
                             continue
-                        principal_type = principal['type']
-                        principal_identifiers = principal['identifiers']
-                        if isinstance(principal_type, list) and len(
-                                principal_type) and 'Federated' in principal_type and isinstance(principal_identifiers,
-                                                                                                 list):
+                        principal_type = principal["type"]
+                        principal_identifiers = principal["identifiers"]
+                        if (
+                            isinstance(principal_type, list)
+                            and len(principal_type)
+                            and "Federated" in principal_type
+                            and isinstance(principal_identifiers, list)
+                        ):
                             for identifier in principal_identifiers:
-                                if isinstance(identifier,
-                                              list) and identifier[0] is not None and \
-                                        'oidc-provider/token.actions.githubusercontent.com' in identifier[0]:
+                                if (
+                                    isinstance(identifier, list)
+                                    and identifier[0] is not None
```

Comment:
Is this access safe? It seems it would be better to check if identifier isn't empty

Reply:
good catch! fixing